conans/client/build/cmake.py

```diff
@@ -233,12 +233,18 @@ def install(self, args=None, build_dir=None):
                                  "defined" % cmake_install_prefix_var_name)
         self._build(args=args, build_dir=build_dir, target="install")
 
-    def test(self, args=None, build_dir=None, target=None):
+    def test(self, args=None, build_dir=None, target=None, progress=False, output_on_failure=False):
         if not self._conanfile.should_test:
             return
         if not target:
             target = "RUN_TESTS" if self.is_multi_configuration else "test"
-        self._build(args=args, build_dir=build_dir, target=target)
+
+        env = {'CTEST_PROGRESS_OUTPUT': '1' if progress else '0',
+               'CTEST_OUTPUT_ON_FAILURE': '1' if output_on_failure else '0'}
+        if self.parallel:
+            env['CTEST_PARALLEL_LEVEL'] = str(cpu_count(self._conanfile.output))
+        with tools.environment_append(env):
```

Comment:
Problem with this approach, I don't think CMake is actually using variables from the environment, so this won't work in practice?

Reply:
ctest does, but I will double-check to make sure and let you know

Reply:
I've just checked with sandbox project, and it worked like a charm

Reply:
It is possible to write a short test, as small as possible, that actually runs ctest?

I also think that it would be very useful to document that in the conan docs too. There is no CTest example or reference at all in the docs.

Reply:
I will definitely add docs
not sure about how to write unit test in that case
how does it work under the hood - cmake generates target called `RUN_TESTS`, which actually executes `ctest` with given arguments. we just invoke `cmake --build . --target RUN_TESTS`, and it spawns `ctest` internally. so it's actually process tree like `cmake` -> `msbuild` -> `ctest`, or `cmake` -> `make` -> `ctest`. as environment variables are inherited, ctest got everything set up.